components/Socket.tsx

```diff
@@ -0,0 +1,91 @@
+import EventEmitter from "stream";
+
+type SocketData = {
+  op: number;
+  data?: object | string;
+  heartbeat?: number;
+};
+
+export const Opcodes = {
+  OPEN: 0, // Socket is opened
+  READY: 1, // Socket is ready
+  DATA: 2, // Socket received data
+  INPUT: 3, // Socket is sending a game input
+  TOGGLE: 4, // Socket is sending a game state toggle command
+  RESTART: 5, // Socket is sending a restart command
+  HEARTBEAT: 10, // Socket is sending a heartbeat
+};
+
+export class TetrisSocket extends EventEmitter {
+  private socket: WebSocket | null;
+
+  private heartbeat: NodeJS.Timer | null;
+
+  constructor() {
+    super();
+
+    this.socket = null;
+    this.heartbeat = null;
+
+    this.init();
+  }
+
+  /**
+   * @brief: init: This function initializes a new Tetris client socket by
+   * setting up socket events and forwarding them to the Tetris React component.
+   */
+  init() {
+    if (this.socket) return;
+
+    this.socket = new WebSocket("ws://localhost:8080");
```

Comment:
I think it would be nice if we can do like a string builder here and maybe define `localhost` and `8080` as URL and port global vars